### PR TITLE
Use reporter.Interface in cmd/cloud

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-Off-By: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: pkg/subctl/cmd/cloud/utils/reporter.go [hunk 1]
```diff
@@ -19,36 +19,30 @@ limitations under the License.
...
 }
 
-func (r *cliReporter) Succeeded(message string, args ...interface{}) {
-	if message != "" {
-		r.status.QueueSuccessMessage(fmt.Sprintf(message, args...))
-	}
-
-	r.status.EndWith(cli.Success)
+func (s *statusReporter) Succeeded(message string, args ...interface{}) {
+	s.status.Success(message, args...)
```
Comment (skitt) [inline pkg/subctl/cmd/cloud/utils/reporter.go:40]: Please don’t drop the check against `message`’s contents.

File: pkg/subctl/cmd/cloud/utils/reporter.go [hunk 2]
```diff
@@ -19,36 +19,30 @@ limitations under the License.
...
 }
 
-func (r *cliReporter) Failed(err ...error) {
-	if len(err) > 0 {
-		r.status.QueueFailureMessage(err[0].Error())
-	}
-
-	r.status.EndWith(cli.Failure)
+func (s *statusReporter) Failed(err ...error) {
+	s.status.Failure(err[0].Error())
```
Comment (skitt) [inline pkg/subctl/cmd/cloud/utils/reporter.go]: Same here, please don’t drop the check against `err`’s contents.

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/subctl/cmd/cloud/utils/reporter.go (hunk 1 above)
Comment (Jaanki) [reply pkg/subctl/cmd/cloud/utils/reporter.go:40]: `status.Success` checks for nil message https://github.com/submariner-io/submariner-operator/blob/devel/internal/cli/status.go#L227 so removed the check here.

File: pkg/subctl/cmd/cloud/utils/reporter.go (hunk 2 above)
Comment (Jaanki) [reply pkg/subctl/cmd/cloud/utils/reporter.go]: Ditto https://github.com/submariner-io/submariner-operator/blob/devel/internal/cli/status.go#L213

Comment (skitt) [reply pkg/subctl/cmd/cloud/utils/reporter.go]: Not quite, if `err` is empty here, `err[0].Error()` will panic.

**Review by skitt — CHANGES_REQUESTED** [review]

**Review by skitt — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**